src/ApplicationManager.cpp

```
@@ -843,6 +843,20 @@ void ApplicationManager::handleSpoolmanSynced(const AppMessage& msg) {
         msg.payload.spoolmanSynced.success ? "true" : "false",
         msg.payload.spoolmanSynced.spoolman_id);
 
+    // Publish resolved spoolman_id to Klipper for toolchanger auto-assign — generic UID tags only.
+    // Smart tags (TigerTag, OpenTag3D, OpenSpool, OpenPrintTag) carry their own data and follow
+    // a separate write/enrichment flow; we deliberately skip the auto-publish there.
+#ifndef NATIVE_TEST
+    if (msg.payload.spoolmanSynced.success && msg.payload.spoolmanSynced.is_uid_lookup
```

Clang (14.0.6)

[note] +1, including nesting penalty of 0, nesting level increased to 1

(clang)